### PR TITLE
Fix context memory manager unit test

### PR DESCRIPTION
**Pull request by 4tXJ7f**
Commit 83f150c727f197c530d6f46a75b516eea52bed29 changed the flag
that we use to determine whether to use the context memory manager
or the debug version. However, the change was not reflected in
the context memory manager unit test. This commit fixes the unit
test to check the correct flag.

### CODE REVIEW

File: test/unit/context/context_mm_black.h [hunk 1]
```diff
@@ -40,7 +40,9 @@ class ContextBlack : public CxxTest::TestSuite {
   }
 
   void testPushPop() {
-#ifdef CVC4_CONTEXT_MEMORY_MANAGER
+#ifdef CVC4_DEBUG_CONTEXT_MEMORY_MANAGER
+#warning "Using the debug context memory manager, omitting unit tests"
+#else /* CVC4_DEBUG_CONTEXT_MEMORY_MANAGER */
```
Comment (aniemetz) [inline test/unit/context/context_mm_black.h]: I find the comment here a little confusing, it kinda implies that the code below is enabled when `CVC4_DEBUG_CONTEXT_MEMORY_MANAGER` is defined.

**Comment by 4tXJ7f** [comment]
This can't happen anymore, the flag (CVC4_CONTEXT_MEMORY_MANAGER) has been replaced with a flag for the opposite (CVC4_DEBUG_CONTEXT_MEMORY_MANAGER).

**Comment by 4tXJ7f** [comment]
I looked at Tim's recent macro definitions in `cvc4_check.{h,cpp}` and he didn't have a comment for the `#else`, so I've decided to remove the comment.

**Comment by aniemetz** [comment]
:+1:

**Pull request merged**